xarray/core/indexing.py

```diff
@@ -800,28 +809,28 @@ def get_duck_array(self):
         self._ensure_cached()
         return self.array.get_duck_array()
 
-    def _oindex_get(self, key):
-        return type(self)(_wrap_numpy_scalars(self.array.oindex[key]))
+    def _oindex_get(self, indexer: OuterIndexer):
+        return type(self)(_wrap_numpy_scalars(self.array.oindex[indexer]))
 
-    def _vindex_get(self, key):
-        return type(self)(_wrap_numpy_scalars(self.array.vindex[key]))
+    def _vindex_get(self, indexer: VectorizedIndexer):
+        return type(self)(_wrap_numpy_scalars(self.array.vindex[indexer]))
 
-    def __getitem__(self, key):
-        self._check_and_raise_if_non_basic_indexer(key)
-        return type(self)(_wrap_numpy_scalars(self.array[key]))
+    def __getitem__(self, indexer: ExplicitIndexer):
```

Comment:
I missed that we were still passing `indexer` objects here. These arrays should not know about `*Indexer`, they should simply receive `*Indexer.tuple` to the approipriate `oindex`, `vindex` or `getitem`

Reply:
Yeah, we haven't made the change so far. I'm happy to sort this out before migrating the indexing functionality to namedarray